### PR TITLE
Changes to migrate to Artifactory

### PR DESCRIPTION
**Pull request by mohnishbasha**
Move kafka build to Artifactory

Use Gradle Artifactory Credentials.

Kafka Team, please help cherry pick this into other branches once PR build passes and merged into master

### CODE REVIEW

**Comment by cmccabe** [comment]
This doesn't look like a complete fix.  We need to remove the other references to nexus in the build files, right?

```
[cmccabe@zeratul ce-kafka]$ git grep nexus
Jenkinsfile:            stage("Publish to nexus") {
Jenkinsfile:                def mavenUrl = 'https://nexus.confluent.io/repository/maven-snapshots/'
build.gradle:      name 'confluent-nexus'
build.gradle:      url 'https://nexus.confluent.io/repository/maven-public'
mk-include/cc-node.mk:NPM_REGISTRY ?= https://nexus.confluent.io/repository/npm-internal/
mk-include/cc-node.mk:  @echo "https://nexus.confluent.io not logged in, Username, Password, and Email not found in environment, prompting for login:" && \
mk-include/cc-node.mk:          npm login --registry=https://nexus.confluent.io/repository/npm-internal/ --scope=@confluent
```

File: Jenkinsfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ def job = {
     }
 
     if (config.publish && config.isDevJob) {
-      configFileProvider([configFile(fileId: 'Gradle Nexus Settings', variable: 'GRADLE_NEXUS_SETTINGS')]) {
+      configFileProvider([configFile(fileId: 'Gradle-Artifactory-Settings', variable: 'GRADLE_NEXUS_SETTINGS')]) {
```
Comment (soondenana) [inline Jenkinsfile]: Do we need to rename `GRADLE_NEXUS_SETTINGS`?

**Comment by mohnishbasha** [comment]
> This doesn't look like a complete fix. We need to remove the other references to nexus in the build files, right?
> 
> ```
> [cmccabe@zeratul ce-kafka]$ git grep nexus
> Jenkinsfile:            stage("Publish to nexus") {
> Jenkinsfile:                def mavenUrl = 'https://nexus.confluent.io/repository/maven-snapshots/'
> build.gradle:      name 'confluent-nexus'
> build.gradle:      url 'https://nexus.confluent.io/repository/maven-public'
> mk-include/cc-node.mk:NPM_REGISTRY ?= https://nexus.confluent.io/repository/npm-internal/
> mk-include/cc-node.mk:  @echo "https://nexus.confluent.io not logged in, Username, Password, and Email not found in environment, prompting for login:" && \
> mk-include/cc-node.mk:          npm login --registry=https://nexus.confluent.io/repository/npm-internal/ --scope=@confluent
> ```

These are only true for ce-kafka and not for kafka.
ce-kafka changes are available at this pr:
https://github.com/confluentinc/ce-kafka/pull/1211

**Comment by rondagostino** [comment]
https://github.com/confluentinc/ce-kafka/pull/1211 is superseded by https://github.com/confluentinc/ce-kafka/pull/1213.

**Comment by mohnishbasha** [comment]
will close this out.

**Comment by mohnishbasha** [comment]
> [confluentinc/ce-kafka#1211](https://github.com/confluentinc/ce-kafka/pull/1211) is superseded by [confluentinc/ce-kafka#1213](https://github.com/confluentinc/ce-kafka/pull/1213).

Thanks, will close this out.

File: Jenkinsfile (hunk 1 above)
Comment (mohnishbasha) [reply Jenkinsfile]: i think we can leave it as is for now, i believe the variable is being used in downstream scripts and configs.

**Pull request merged**